### PR TITLE
hotfix: undo our modifications to the Python build system

### PR DESCRIPTION
**Pull request by matz-e**
These changes clash with upstream modifications, leading to the prefix
being *twice* in the installation directories / nesting way too deep.

Thanks @olupton for the prod!

### CODE REVIEW

File: lib/spack/spack/build_systems/python.py [hunk 1]
```diff
@@ -258,7 +252,7 @@ def install_args(self, spec, prefix):
                  '--install-purelib=%s' % pure_site_packages_dir,
                  '--install-platlib=%s' % plat_site_packages_dir,
                  '--install-scripts=bin',
-                 '--install-data=""',
+                 '--install-data=',
```
Comment (olupton) [inline lib/spack/spack/build_systems/python.py:255]: Why this change? This seems to be the only diff to upstream: https://github.com/BlueBrain/spack/compare/hotfix-python-install..spack:releases/v0.16#diff-61e125b04c4e9783dda6c50777b7a9ccb298b342a1275a6df8fd736a1f5dc829

**Review by olupton — APPROVED** [review]
LGTM, one trivial question

File: lib/spack/spack/build_systems/python.py (hunk 1 above)
Comment (matz-e) [reply lib/spack/spack/build_systems/python.py:255]: Because on at least my shell, this gives a literal `""/share/…` in the install prefix instead of just `share/…`

Comment (matz-e) [reply lib/spack/spack/build_systems/python.py:255]: See also: https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/python.py#L266

Comment (olupton) [reply lib/spack/spack/build_systems/python.py:255]: OK!
